File: python/01_binary_search.py
def search_binary(list, elem):
  low = 0
  high = len(list) - 1
  mid = 0

  while low <= high:

    mid=(low+high) // 2

    if list[mid] == elem:
      return mid
    
    if list[mid] > elem:
      high = mid - 1
    
    else:
      low = mid + 1

  return -1;

my_list=[1, 3, 5, 7, 9]

print(search_binary(my_list, 3))
print(search_binary(my_list, -1))

File: typescript/01_binary_search.ts
function searchBinary(list: Array<number>, elem: number) {
  let low: number = 0;
  let high: number = list.length - 1;
  
  
  while (low <= high) {
     const medium: number = (low + high) / 2;
     let estimate: number = list[Math.round(medium)];    
      if (estimate === elem) {
        return estimate;
      }
    
      if (estimate > elem) {
        high = medium - 1;
      } else {
        low = medium + 1;
      }
  }
  
  return null;

}

const mylist: Array<number> = [1, 3, 5, 7, 9];
  
console.log(searchBinary(mylist, 3));
console.log(searchBinary(mylist, -1));